src/main/java/seedu/address/model/queue/QueueManager.java

```diff
@@ -77,10 +77,26 @@ public void poll() {
         queueList.remove(0);
     }
 
-    public boolean hasId(ReferenceId id) {
+    public boolean hasIdInQueue(ReferenceId id) {
         return queueList.contains(id);
     }
 
+    /**
+     * Checks if a patient is being served
+     *
+     * @param id of the patient being served
+     * @return boolean which indicates whether the patient is being served
+     */
+    public boolean hasIdInRooms(ReferenceId id) {
+        UniqueElementList<ReferenceId> patientsBeingServed = new UniqueElementList<>();
```

Comment:
Please rewrite this function. Why are you constructing another uniqueElementList?

Reply:
Ok changed.